src/lib.ts

```diff
@@ -377,23 +390,32 @@ export function getZapGroup(item: Item): Item[] {
 
 /**
  * Get a map of banished monsters keyed by what banished them
- * 
+ *
  * @category General
  */
 export function getBanishedMonsters(): Map<Item | Skill, Monster> {
-  const banishedstring = get("banishedMonsters");
-  const banishedComponents = banishedstring.split(":");
+  const banishes = chunk(get("banishedMonsters").split(":"), 3);
+
   const result = new Map<Item | Skill, Monster>();
-  if (banishedComponents.length < 3) return result;
-  for (let idx = 0; idx < banishedComponents.length / 3 - 1; idx++) {
-    const foe = Monster.get(banishedComponents[idx * 3]);
-    const banisher = banishedComponents[idx * 3 + 1];
+
+  for (const [foe, banisher] of banishes) {
+    if (foe === undefined || banisher === undefined) break;
     // toItem doesn"t error if the item doesn"t exist, so we have to use that.
-    const banisherItem = Item.get(banisher);
+    const banisherItem = toItem(banisher);
```

Comment:
@phulin sorry, I somehow ignored the comment above this when I moved this function, but I've fixed it now. And made it a little nicer imo imo